### PR TITLE
Thread typing indicator

### PR DESCRIPTION
**Pull request by yetieaterxb1**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request

Moves TypingIndicator to MessageList and adjusts TypingIndicator to only show in thread or main channel

### CODE REVIEW

File: src/components/TypingIndicator/__tests__/TypingIndicator.test.js [hunk 1]
```diff
@@ -139,4 +139,120 @@ describe('TypingIndicator', () => {
...
+            example: { parent_id: 'sample-thread', user: 'test-user' },
+          },
+          channel,
+          thread: { id: 'sample-thread' },
+        }}
+      >
+        <TypingIndicator threadList />
+      </ChannelContext.Provider>,
+    );
+    console.log({ container });
```
Comment (DanC5) [inline src/components/TypingIndicator/__tests__/TypingIndicator.test.js:164]: remove console logs

File: src/components/TypingIndicator/__tests__/TypingIndicator.test.js [hunk 2]
```diff
@@ -139,4 +139,120 @@ describe('TypingIndicator', () => {
...
   });
+
+  it('should render TypingIndicator if user is typing in thread', async () => {
+    const client = await getTestClientWithUser();
+    const ch = generateChannel({ config: { typing_events: true } });
+    useMockedApis(client, [getOrCreateChannelApi(ch)]);
+    const channel = client.channel('messaging', ch.id);
+    await channel.watch();
+
+    const { container } = render(
```
Comment (DanC5) [inline src/components/TypingIndicator/__tests__/TypingIndicator.test.js:150]: let's pull the render logic out of the tests and put in the beginning of the file and name it something like renderTypingIndicator and pass values needed for each test into it, so don't need to repeat this logic so much

File: src/components/MessageList/MessageList.js [hunk 3]
```diff
@@ -18,7 +18,10 @@ import { EmptyStateIndicator as DefaultEmptyStateIndicator } from '../EmptyState
 import { LoadingIndicator as DefaultLoadingIndicator } from '../Loading';
 import { EventComponent } from '../EventComponent';
 import { DateSeparator as DefaultDateSeparator } from '../DateSeparator';
-import { TypingIndicator as DefaultTypingIndicator } from '../TypingIndicator';
+import {
+  TypingIndicator as DefaultTypingIndicator,
```
Comment (DanC5) [inline src/components/MessageList/MessageList.js:22]: can delete, adjust prop types at bottom

File: src/components/TypingIndicator/TypingIndicator.js [hunk 4]
```diff
@@ -9,20 +9,30 @@ import { Avatar as DefaultAvatar } from '../Avatar';
...
-const TypingIndicator = ({ Avatar = DefaultAvatar, avatarSize = 32 }) => {
-  const { typing, client, channel } = useContext(ChannelContext);
+const TypingIndicator = ({
+  Avatar = DefaultAvatar,
+  avatarSize = 32,
+  threadList,
+}) => {
+  const { typing, client, channel, thread } = useContext(ChannelContext);
 
   if (!typing || !client || channel?.getConfig()?.typing_events === false)
```
Comment (DanC5) [inline src/components/TypingIndicator/TypingIndicator.js:19]: brackets

File: src/components/TypingIndicator/TypingIndicator.js [hunk 5]
```diff
@@ -9,20 +9,30 @@ import { Avatar as DefaultAvatar } from '../Avatar';
...
     return null;
 
   const users = Object.values(typing).filter(
```
Comment (DanC5) [inline src/components/TypingIndicator/TypingIndicator.js:22]: typingInChannel or something similar

**Review by DanC5 — APPROVED** [review]

**Review by nhannah — APPROVED** [review]

**Pull request merged**